component/stream/stream.go

```diff
@@ -134,104 +117,52 @@ func (st *Stream) Configure() error {
+		if !conf.readOnly {
 			log.Err("Stream::Configure : config error, handle level caching is available for read-only mode")
 			return errors.New("handle level caching is available for read-only mode")
 		}
-
-		st.cacheOnFileHandle = conf.CacheOnFileHandle
-		maxBlocks := int(math.Floor(float64(conf.StreamCacheSize) / float64(conf.BlockSize)))
-
-		// default eviction policy is LRU
-		var evictionPolicy common.EvictionPolicy
-		evictionPolicy.Parse(strings.ToLower(conf.Policy))
-
-		var bc gcache.Cache
-		var diskBlockCache gcache.Cache
-
-		switch evictionPolicy {
-		case common.EPolicy.LFU():
-			bc = gcache.New(maxBlocks).LFU().EvictedFunc(st.evictBlock).Build()
-		case common.EPolicy.ARC():
-			bc = gcache.New(maxBlocks).ARC().EvictedFunc(st.evictBlock).Build()
-		default:
-			bc = gcache.New(maxBlocks).LRU().EvictedFunc(st.evictBlock).Build()
-		}
-		log.Trace("Stream::Configure : cache eviction policy %s", evictionPolicy)
-
-		if conf.DiskPersistence {
-			maxDiskBlocks := int(math.Floor(float64(conf.DiskCacheSize) / float64(conf.BlockSize)))
-			diskBlockCache = gcache.New(maxDiskBlocks).LRU().EvictedFunc(st.evictDiskBlock).Build()
-
-			if conf.DiskTimeoutSec == 0 {
-				conf.DiskTimeoutSec = defaultDiskTimeoutSec
-			}
-
-			if conf.DiskPath == "" {
-				log.Err("Stream::Configure : Config error [disk-cache-path not set]")
-				return fmt.Errorf("config error in %s [disk-cache-path not set]", st.Name())
-			}
-
-			_, err = os.Stat(conf.DiskPath)
-			if os.IsNotExist(err) {
-				log.Info("Stream::Configure : Config error [disk-cache-path does not exist. attempting to create]")
-				err := os.Mkdir(conf.DiskPath, os.FileMode(0755))
-				if err != nil {
-					log.Err("Stream::Configure : Config error creating temp directory failed [%s]", err.Error())
-					return fmt.Errorf("failed to create temp directory for stream persistence [%s]", err.Error())
-				}
-			}
-		}
-
-		st.streamCache = &cache{
-			files:            make(map[string]*cacheFile),
-			blockSize:        int64(conf.BlockSize) * mb,
-			maxBlocks:        maxBlocks,
-			blocksPerFileKey: conf.BlocksPerFile,
-			blocks:           bc,
-			evictionPolicy:   evictionPolicy,
-			diskPersistence:  conf.DiskPersistence,
-			diskPath:         conf.DiskPath,
-			diskCacheMB:      int64(conf.DiskCacheSize),
-			diskTimeoutSec:   float64(conf.DiskTimeoutSec),
-			diskBlocks:       diskBlockCache,
-		}
+	}
+	st.cache = &cache{
+		blockSize:           int64(conf.BlockSize) * mb,
+		bufferSizePerHandle: conf.BufferSizePerFile * mb,
+		handleLimit:         int32(conf.HandleLimit),
+		openHandles:         0,
 	}
 	return nil
 }
 
 // Stop : Stop the component functionality and kill all threads started
 func (st *Stream) Stop() error {
 	log.Trace("Stopping component : %s", st.Name())
-	if !st.streamOnly {
-		st.streamCache.teardown()
+	handleMap := handlemap.GetHandles()
+	handles := map[string]interface{}{}
+	handleMap.Range(func(key, value interface{}) bool {
```

Comment:
Why there are two iterations one to convert handle map to a slice and then iterate a slice. This could have been done in single iteration as well.